hazelcast/src/main/java/com/hazelcast/cache/impl/AbstractHazelcastCachingProvider.java

```diff
@@ -386,4 +386,25 @@ private boolean isConfigLocation(URI location) {
         }
         return (scheme != null && SUPPORTED_SCHEMES.contains(scheme.toLowerCase(StringUtil.LOCALE_INTERNAL)));
     }
+
+    /**
+     * Tests if the configuration file provided is XML. If the test fails then it's assumed to be YAML. The test is based on
+     * the file extension of the configuration file.
+     * @param configURL Configuration URL to test.
+     * @throws IllegalArgumentException if the file extension is not one of the following: .xml, .yml or .yaml.
+     * @return true if the configuration file is XML, otherwise false.
+     */
+    protected static boolean isXmlConfiguration(URL configURL) {
+        String path = configURL.getPath();
+        if (path.endsWith(".xml")) {
+            return true;
+        }
+
+        if (path.endsWith(".yml") || path.endsWith(".yaml")) {
+            return false;
+        }
+
+        throw new IllegalArgumentException(
```

Comment:
Backward compatibility consideration: in previous versions, having a `.xml` extension was not a requirement for configuration URLs. Now a config URL that does not end with `.xml` will fail with an exception.

So there are two options:
- break non-`.xml` parsing, add a note to release notes so people might notice before upgrading.
- assume XML format for any URL path that does not match `.yaml`/`.yml`. This way we don't risk breaking existing working setups. In any case, if not a valid XML file, parsing will fail a bit further down the road.

wdyt?

Reply:
@vbekiaris ah, your second point seems to be best approach IMO: that will satisfy the new YAML piece while being backwards compatible.

If you're OK with that I will go ahead and make the change.